setup.py

```diff
@@ -364,6 +363,11 @@ def clean_up(self):
         shutil.rmtree(self.build_temp)
 
     def run(self):
+        # `is_local` requires running the local hadoop executable.
+        # Don't move this call into other methods of the class that
+        # may be called while executing other commands (e.g., clean)
+        if HADOOP_VERSION_INFO.is_local():
+            exit('Build failed, local mode is not supported.')
```

Comment:
I'd change this line to `raise pydoop.LocalModeNotSupported()`, as in other modules. The error message is more informative and the traceback is a better indication that something went wrong (especially when Pydoop installation is part of a longer process, such as a Docker image build).